### ISSUE TITLE
Reproducible train/test/val splits

### ISSUE THREAD

**Issue by eroell** [WenjieDu/BenchPOTS#21]
### 1. Feature description

New argument `random_state` in `preprocess_<dataset>`:

`benchpots.datasets.preprocess_physionet2012(subset="all", rate=0, random_state=None)`

Not 100% sure that this is the only place where a seed would be needed s.t. `preprocess_<dataset>` generate reproducible dataset splits btw!

### 2. Motivation

Reproducibility.
Right now, there is no way of seeding sklearn's `train_test_split` e.g. here
https://github.com/WenjieDu/BenchPOTS/blob/f7d4c97d47c20e19adfff411a7fe4dcd95fb23b3/benchpots/datasets/physionet_2012.py#L124

This leads to new splits whenever a dataset preprocessing is done...

### 3. Your contribution

Can do PR, but might take a while?
